packages/flutter_tools/lib/src/bundle.dart

```diff
@@ -84,34 +84,83 @@ class BundleBuilder {
   /// The default  `manifestPath` is `pubspec.yaml`
   Future<void> build({
```

Comment:
You could arrange this a little bit differently to make the unit tests a bit easier to write for now (avoiding the top level `env` declaration.

Add a new optional parameter: `@visibleForTesting BuildSystem buildSystemOverride`

Then use it with something like `buildSystemOverride ?? globals.buildSystem`. You can then pass the build system directly to the bundle builder

Reply:
done